### PR TITLE
[pybind11] Update to v2.5.0

### LINKED ISSUE #12371: [pybind11] update to  2.5.0
Library name: pybind11

New version number: v2.5.0

### PR DESCRIPTION
**Pull request by anand-bala**
**Describe the pull request**

Updates pybind11 to the latest stable version v2.5.0 from v2.4.3.

- What does your PR fix? Fixes #12371 

- Which triplets are supported/not supported? Have you updated the CI baseline?

Supports all the previously supported triplets (all), but have checked on personal x64-linux and x64-windows. No changed to the CI baseline.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Yes.

### CODE REVIEW

File: ports/pybind11/portfile.cmake [hunk 1]
```diff
@@ -3,8 +3,8 @@ include(vcpkg_common_functions)
 vcpkg_from_github(
```
Comment (NancyLi1013) [inline ports/pybind11/portfile.cmake:3]: `include(vcpkg_common_functions)` is no longer needed here. Could you please remove it?

File: ports/pybind11/portfile.cmake [hunk 2]
```diff
@@ -3,8 +3,8 @@ include(vcpkg_common_functions)
...
     OUT_SOURCE_PATH SOURCE_PATH
     REPO pybind/pybind11
-    REF 80d452484c5409444b0ec19383faa84bb7a4d351 # v2.4.3
-    SHA512 987f8c075ff3e4f90ab27a6121f3767a82939e35cd2143649819c8d39b09d1c234d39fa204ed5f6bd1d9ec97c275f590df358769d7726a16ccb720a91c550883
+    REF 3b1dbebabc801c9cf6f0953a4c20b904d444f879 # v2.5.0
+    SHA512 1a75d29447dbba96eebf8ecdebad1be0dd5327c32f5122b0ece9d9ec22eae4feacd0efb3a5070b3a135a892b1682c7215b0c529b179493694df932945a379f4c
     HEAD_REF master
 )
```
Comment (NancyLi1013) [inline ports/pybind11/portfile.cmake:10]: We prefer to use[ vcpkg_add_to_path()](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_add_to_path.md) to add a directory to the PATH environment variable. Could you please update it?

File: ports/pybind11/portfile.cmake [hunk 3]
```diff
@@ -34,4 +34,4 @@ vcpkg_install_cmake()
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/)
 
 # copy license
-file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/pybind11/copyright)
\ No newline at end of file
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/pybind11/copyright)
```
Comment (NancyLi1013) [inline ports/pybind11/portfile.cmake:37]:
```suggestion
file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @anand-bala 
Thanks for this PR.
Since `fmt `has been updated in this PR #12312. Could you please revert the changes for `fmt`?

**Comment by anand-bala** [comment]
Hey @NancyLi1013, I cleaned up the fmt changes and made the edits you suggested.

**Comment by ras0219-msft** [comment]
LGTM, thanks for the update!

**Pull request merged**